src/main/java/org/schemaspy/LayoutFiles.java

```diff
@@ -15,21 +11,17 @@
 public class LayoutFiles {
 
     private final ClassLoader classLoader;
-    private final FileFilter filter;
-    private final File outputDir;
 
-    public LayoutFiles(ClassLoader classLoader, FileFilter filter, File outputDir) {
+    public LayoutFiles(ClassLoader classLoader) {
         this.classLoader = classLoader;
-        this.filter = filter;
-        this.outputDir = outputDir;
     }
 
     /**
-     * Copy the files - sans template .html files - to object's destination.
+     * The layout files - sans template .html files.
      *
-     * @throws IOException when not possible to copy layout files to outputDir
+     * @throws IOException when not possible to retrieve layout files
      */
-    public void prepare() throws IOException {
+    public URL files() throws IOException {
```

Comment:
It's strange with `files()`and only returning a single URL.

I'm guessing we probably should be talking about `LayoutFolder`

Reply:
If correct, then we can remove most or all of `ResourceWriter`

Reply:
My issue was name implies plural, return is singular.

Would still be a URL just that the method would be folder and the object be a LayoutFolder. Not sure how that would impact ResourceWriter.

As mentioned in another PR, one responsibility would be to provide a collection of files, one filter that collection and a third writing them. ResourceWriter has lots of responsibilities now.

Reply:
If you check the usages of `ResourceWriter` you will find none other than with `LayoutFiles` even though it has code for handling a file, a folder and jar i.e., we have dead code.